x-pack/plugins/fleet/server/services/setup.ts

```diff
@@ -145,21 +150,43 @@ export async function ensureFleetGlobalEsAssets(
   // Ensure Global Fleet ES assets are installed
   logger.debug('Creating Fleet component template and ingest pipeline');
   const globalAssetsRes = await Promise.all([
-    ensureDefaultComponentTemplate(esClient, logger),
+    ensureDefaultComponentTemplates(esClient, logger), // returns an array
     ensureFleetFinalPipelineIsInstalled(esClient, logger),
   ]);
-
-  if (globalAssetsRes.some((asset) => asset.isCreated)) {
+  const assetResults = globalAssetsRes.flat();
+  if (assetResults.some((asset) => asset.isCreated)) {
     // Update existing index template
-    const packages = await getInstallations(soClient);
-
+    const installedPackages = await getInstallations(soClient);
+    const bundledPackages = await getBundledPackages();
+    const findMatchingBundledPkg = (pkg: Installation) =>
+      bundledPackages.find(
+        (bundledPkg: BundledPackage) =>
+          bundledPkg.name === pkg.name && bundledPkg.version === pkg.version
+      );
     await Promise.all(
-      packages.saved_objects.map(async ({ attributes: installation }) => {
+      installedPackages.saved_objects.map(async ({ attributes: installation }) => {
```

Comment:
I started writing this but the amount of mocking was getting too much, I think we could restructure the code to better allow for unit testing of the install logic but it may be worth a tech debt ticket?